api/envoy/config/rbac/v3/rbac.proto

```diff
@@ -270,12 +270,16 @@ message Permission {
     // Extension for configuring custom matchers for RBAC.
     // [#extension-category: envoy.rbac.matchers]
     core.v3.TypedExtensionConfig matcher = 12;
```

Comment:
we already have typed matchers, seems like adding another typedextensionconfig y kind of redundant? can't we simply create a matcher for `uri_template`?

Reply:
I don't understand the extension system well - does it mean that I would have to copy [uri_template_match.proto](https://github.com/envoyproxy/envoy/blob/main/api/envoy/extensions/path/match/uri_template/v3/uri_template_match.proto) and paste it next to [upstream_ip_port_matcher.proto](https://github.com/envoyproxy/envoy/tree/main/api/envoy/extensions/rbac/matchers/upstream_ip_port/v3) while adjusting the `// [#extension: envoy.path.match.uri_template.uri_template_matcher]` to also begin with `envoy.rbac.matchers` as in `upstream_ip_port_matcher.proto`?

Reply:
i did as i described in the comment

Reply:
No, we should not copy+paste like that. Before rolling back, @sschepens are you suggesting we just used the generic matcher capability here and plug into that?

I don't think we need a static entry for URI template, I don't think we need to grow the static match capabilities here, rather we do it via generic match on URL (as I was proposing and was present in https://github.com/envoyproxy/envoy/pull/31447/commits/528c89773c456a2e4f397a17ff4edb2a39b7fb6f, or we make URI template matching a generic matcher extension as well.

Reply:
> No, we should not copy+paste like that. Before rolling back, @sschepens are you suggesting we just used the generic matcher capability here and plug into that?
> 
> I don't think we need a static entry for URI template, I don't think we need to grow the static match capabilities here, rather we do it via generic match on URL (as I was proposing and was present in [528c897](https://github.com/envoyproxy/envoy/commit/528c89773c456a2e4f397a17ff4edb2a39b7fb6f), or we make URI template matching a generic matcher extension as well.

In all honesty I don't really like extensions since they make for a ver verbose usage, more so when we can have many matchers. I'd really like for URI matching to become a first class and probably be included without needing `TypedExtensionConfig`, but  I don't think that's going to happen. So, for me, it made more sense to make URI template a generic matcher extension and avoid adding another member to PathMatcher making use of the standard matcher extension mechanism.

Reply:
I'd suggest opening a discussions with @envoyproxy/api-shepherds on the tradeoffs between extensibility and verbosity in configuration - in general we err towards more pluggable extension points and assume for the most part machine generation of configuration. xDS is sometimes read/written by humans, but we do not have any goal today around conciseness, in particular when it conflicts with generic extension points.

In this case, URI matching is a plugin for an existing extension class, so it seem arbitrary to opt to hardcode it into the RBAC API rather than the extension class.